contracts/contracts/strategies/amo/ConvexEthMetaStrategy.sol

```diff
@@ -0,0 +1,171 @@
+    address private _deprecated_weth;
+
+    // Ordered list of pool assets
+    // slither-disable-next-line constable-states
+    uint128 private _deprecated_oethCoinIndex = 1;
+    // slither-disable-next-line constable-states
+    uint128 private _deprecated_ethCoinIndex = 0;
+
+    // Added for backward compatibility
+    address public immutable oeth;
+    address public immutable weth;
+
+    constructor(
+        BaseStrategyConfig memory _baseConfig,
+        AMOConfig memory _amoConfig,
+        ConvexConfig memory _convexConfig
+    ) BaseConvexAMOStrategy(_baseConfig, _amoConfig, _convexConfig) {
+        oeth = _amoConfig.oTokenAddress;
+        weth = _amoConfig.assetAddress;
+    }
+
+    /***************************************
+        Vault to Pool Asset Conversions
+    ****************************************/
+
+    /// @dev Unwraps the ETH from WETH using WETH withdraw
+    function _toPoolAsset(address, uint256 wethAmount)
+        internal
+        override
+        returns (uint256 ethAmount)
+    {
+        IWETH9(address(asset)).withdraw(wethAmount);
+        ethAmount = wethAmount;
+    }
+
+    function _calcPoolAsset(address, uint256 wethAmount)
+        internal
+        pure
+        override
+        returns (uint256 ethAmount)
+    {
+        ethAmount = wethAmount;
+    }
+
+    /// @dev The OETH OToken is 1:1 to ETH so return the ETH amount
+    function _toOTokens(uint256 ethAmount)
+        internal
+        pure
+        override
+        returns (uint256 oethAmount)
+    {
+        oethAmount = ethAmount;
+    }
+
+    /***************************************
+            Curve Pool Withdrawals
+    ****************************************/
+
+    function _withdrawAsset(
+        address,
+        uint256 vaultAssetAmount,
+        address recipient
+    ) internal override {
+        // Convert ETH to WETH
+        IWETH9(address(asset)).deposit{ value: vaultAssetAmount }();
+
+        // Transfer the WETH to the Vault
+        require(
+            asset.transfer(recipient, vaultAssetAmount),
+            "WETH transfer failed"
+        );
+
+        emit Withdrawal(address(asset), address(lpToken), vaultAssetAmount);
+    }
+
+    /// @dev Gets the ETH balance of this strategy contract
+    /// and then converts all the ETH to WETH
+    function _withdrawAllAsset(address recipient) internal override {
+        // Get ETH balance of this strategy contract
+        uint256 ethBalance = address(this).balance;
+        if (ethBalance > 0) {
+            _withdrawAsset(address(asset), ethBalance, recipient);
+        }
+    }
+
+    /***************************************
+                Asset Balance
+    ****************************************/
+
+    /**
+     * @notice Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    ETH value of both OETH and ETH in the Curve pool
+     */
+    function checkBalance(address _asset)
```

Comment:
Should include the `checkBalance()` without asset parameters as well?

Reply:
Why would we do this? Doesn't the Vault only work with `checkBalance(address _asset)`?

Reply:
Yes correct it does, I was under the impression though that we will support `checkBalance()` on all future strategies (and at some point implement it for existing ones) to at some point migrate VaultCore to call the checkBalance without the asset parameter